### PR TITLE
Create fixed roles for reading API Keys and service accounts and fix listing of service account tokens

### PR DESCRIPTION
**Pull request by vtorosyan**
**What this PR does / why we need it**:

- The pattern we follow with most of the fixed roles assumes to have two roles for reading and writing. This PR adds two fixed roles for reading API Keys and Service Accounts.
- Additionally, when viewing a service account, listing tokens was failing with an error due to wrong SQL query. This is also fixed in this PR.

Screenshot 2022-04-14 at 13 56 04

**Special notes for your reviewer**:

I am not touching the docs now to avoid conflict with the refactoring, instead creating an issue to add the new roles into the docs.

### CODE REVIEW

File: pkg/services/serviceaccounts/manager/roles.go [hunk 1]
```diff
@@ -6,7 +6,24 @@ import (
 )
 
 func RegisterRoles(ac accesscontrol.AccessControl) error {
-	role := accesscontrol.RoleRegistration{
+	saReader := accesscontrol.RoleRegistration{
+		Role: accesscontrol.RoleDTO{
+			Version:     4,
```
Comment (IevaVasiljeva) [inline pkg/services/serviceaccounts/manager/roles.go:11]: nit: the version should be 1.

File: pkg/services/serviceaccounts/manager/roles.go [hunk 2]
```diff
@@ -6,7 +6,24 @@ import (
...
+			Name:        "fixed:serviceaccounts:reader",
+			DisplayName: "Service accounts reader",
+			Description: "Read service accounts.",
```
Comment (IevaVasiljeva) [inline pkg/services/serviceaccounts/manager/roles.go:14]: nit: maybe 
```suggestion
			Description: "Read service accounts and service account tokens.",
```

File: pkg/services/serviceaccounts/manager/roles.go [hunk 3]
```diff
@@ -6,7 +6,24 @@ import (
...
+			Group:       "Service accounts",
+			Permissions: []accesscontrol.Permission{
+				{
+					Action: serviceaccounts.ActionRead,
+					Scope:  serviceaccounts.ScopeAll,
+				},
+			},
+		},
+		Grants: []string{"Admin"},
```
Comment (IevaVasiljeva) [inline pkg/services/serviceaccounts/manager/roles.go:23]: nit: should we use `string(models.ROLE_ADMIN)` here as well?

**Review by IevaVasiljeva — APPROVED** [review]
Some small suggestions, otherwise looks good.

**Comment by vtorosyan** [comment]
Thanks @IevaVasiljeva , I have updated with your suggestions and additionally included a small bug fix, hope that's alright. Would appreciate review again.

**Review by IevaVasiljeva — APPROVED** [review]
LGTM!

**Pull request merged**